scripts/lib/CIME/case/case_st_archive.py

```diff
@@ -257,36 +256,22 @@ def _archive_history_files(archive, archive_entry,
 
     # archive history files - the only history files that kept in the
     # run directory are those that are needed for restarts
-
-    for suffix in archive.get_hist_file_extensions(archive_entry):
-        if compname.find('mpas') == 0 or compname == 'mali':
-            newsuffix =                    compname + r'\d*'
-        else:
-            newsuffix = casename + r'\.' + compname + r'_?' + r'\d*'
-        newsuffix += r'\.' + suffix
-        if not suffix.endswith('$'):
-            newsuffix += r'\.'
-
-        logger.debug("short term archiving suffix is {} ".format(newsuffix))
-
-        pfile = re.compile(newsuffix)
-        histfiles = [f for f in os.listdir(rundir) if pfile.search(f)]
-        logger.debug("histfiles = {} ".format(histfiles))
```

Comment:
I haven't taken the time to understand this deletion. Given your request for a careful review: Do you want a careful review of this change, or are you pretty confident about this one?

Reply:
I have tested case.st_archive --test-all with CIME_MODEL=e3sm and all pass.